### PR TITLE
Listen on edited pull request

### LINKED ISSUE #86: Listen od "edited" action for the "pull_request" webhook.
We can additionally listen on the "edited" action for the "pull_request" webhook. This action is triggered by changing either the [title, body or the base branch of the PR](https://frontside.com/blog/2020-05-26-github-actions-pull_request/). 

Once an "edited" action is triggered, we can update the title of the PR, and most importantly, the base branch of the PR.  Rejecting merge commands depending on whether or not you are merging to "master", relies on the data being up to date.

### PR DESCRIPTION
**Pull request by adrianotm**
Fixes #86

### CODE REVIEW

File: src/Logic.hs [hunk 1]
```diff
@@ -311,6 +313,10 @@ handlePullRequestClosed pr state = Pr.deletePullRequest pr <$>
         pure state { Pr.integrationCandidate = Nothing }
       _notCandidatePr -> pure state
 
+handlePullRequestEdited :: PullRequestId -> Text -> ProjectState -> Action ProjectState
+handlePullRequestEdited prId newTitle = pure . Pr.updatePullRequest prId changeTitleAndBaseBranch
+  where changeTitleAndBaseBranch pr = pr { Pr.title = newTitle }
```
Comment (FPtje) [inline src/Logic.hs:318]: This name does not yet apply, but it is also not future proof. Maybe rename to `updatePr`?

File: tests/Spec.hs [hunk 2]
```diff
@@ -230,6 +230,13 @@ main = hspec $ do
       state `shouldSatisfy` not . Project.existsPullRequest (PullRequestId 1)
       state `shouldSatisfy` Project.existsPullRequest (PullRequestId 2)
 
+    it "handles PullRequestEdited" $ do
+      let event1 = PullRequestOpened (PullRequestId 1) (Branch "p") (Sha "abc") "title" "peter"
+          event2 = PullRequestEdited (PullRequestId 1) "newTitle"
+          state = fst $ runAction $ handleEventsTest [event1, event2] Project.emptyProjectState
+          pr = fromJust $ Project.lookupPullRequest (PullRequestId 1) state
+      Project.title pr      `shouldBe` "newTitle"
```
Comment (FPtje) [inline tests/Spec.hs:238]: Nit: Odd spacing

**Review by FPtje — CHANGES_REQUESTED** [review]
Nice work! A couple of small nits, but otherwise looks good!

**Review by FPtje — APPROVED** [review]
LGTM!

**Comment by adrianotm** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @adrianotm, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as abbf862e9d9c46c1cce64f4c6fae3fe2e37b112b, waiting for CI …

**Pull request merged**